gi/object.cpp

```diff
@@ -2524,6 +2596,189 @@ gjs_add_interface(GType instance_type,
                                 &interface_vtable);
 }
 
+static gboolean
+validate_interfaces_and_properties_args(JSContext *cx,
+                                        JSObject  *interfaces,
+                                        JSObject  *properties,
+                                        guint32   *n_interfaces,
+                                        guint32   *n_properties)
+{
+    guint32 n_int, n_prop;
+
+    if (!JS_IsArrayObject(cx, interfaces)) {
+        gjs_throw(cx, "Invalid parameter interfaces (expected Array)");
+        return FALSE;
+    }
+
+    if (!JS_GetArrayLength(cx, interfaces, &n_int))
```

Comment:
Hmmm. I actually don't know the answer to this - but is there any reason in particular why JS_GetArrayLength can fail, apart internal inconsistency in the engine itself? If there is one, maybe we should be warning about it, or even just asserting that it returns JS_TRUE. Not sure what to do here :/